plot/plot.app.js

```diff
@@ -1157,16 +1260,73 @@
                     if (newValue) vm.groups = newValue;
                 });
             }])
-            .directive('plot', ['$rootScope', '$timeout', function ($rootScope, $timeout) {
-
+            .directive('plot', ['$rootScope', '$timeout','ConfigUtils', function ($rootScope, $timeout,ConfigUtils) {
+                
                 return {
                     link: function (scope, element) {
                         scope.$watch('plots', function (plots) {
                             if (plots) {
                                 $timeout(function() {
                                     for (var i = 0; i < plots.length; i++) {
                                         if (plots[i].id == element[0].attributes['plot-id'].nodeValue) {
-                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config).then(function () {
+                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config)
```

Comment:
plotly_click is an event that takes place on the click of bar or chart.

plot_legendclick is an event that takes place on click of legends.